Goal: Seems to be multiple:
      - chunk 1: ?
      - chunk 2: ?
      - chunk 3: installing smbmount and smbumount is done The Debian Way in debian/rules

Fixes: ?

Status wrt upstream: Chunk 3 is Debian specific. Not sure for chunks 1 and 2

Note: 

Index: samba-3.0.25a/source/client/smbmount.c
===================================================================
--- samba-3.0.25a.orig/source/client/smbmount.c	2007-05-26 07:45:39.744216228 +0200
+++ samba-3.0.25a/source/client/smbmount.c	2007-05-26 07:46:32.692638041 +0200
@@ -768,7 +768,7 @@
 					*lp = 0;
 					pstrcpy(password,lp+1);
 					got_pass = True;
-					memset(strchr_m(opteq+1,'%')+1,'X',strlen(password));
+					memset(strchr_m(opteq+1,'%')+1,'\0',strlen(password));
 				}
 				if ((lp=strchr_m(username,'/'))) {
 					*lp = 0;
@@ -778,7 +778,7 @@
 				  !strcmp(opts, "password")) {
 				pstrcpy(password,opteq+1);
 				got_pass = True;
-				memset(opteq+1,'X',strlen(password));
+				memset(opteq+1,'\0',strlen(password));
 			} else if(!strcmp(opts, "credentials")) {
 				pstrcpy(credentials,opteq+1);
 			} else if(!strcmp(opts, "netbiosname")) {
@@ -892,7 +892,7 @@
 			*p = 0;
 			pstrcpy(password,p+1);
 			got_pass = True;
-			memset(strchr_m(getenv("USER"),'%')+1,'X',strlen(password));
+			memset(strchr_m(getenv("USER"),'%')+1,'\0',strlen(password));
 		}
 		strupper_m(username);
 	}
Index: samba-3.0.25a/source/web/diagnose.c
===================================================================
--- samba-3.0.25a.orig/source/web/diagnose.c	2007-05-26 07:45:39.752216292 +0200
+++ samba-3.0.25a/source/web/diagnose.c	2007-05-26 07:46:32.696638073 +0200
@@ -62,6 +62,7 @@
 {
 	struct cli_state *cli;
 
+ 	loopback_ip.s_addr = htonl((127 << 24) + 1);
 	if ((cli = cli_initialise()) == NULL)
 		return False;
 
Index: samba-3.0.25a/source/script/installbin.sh.in
===================================================================
--- samba-3.0.25a.orig/source/script/installbin.sh.in	2007-05-26 07:45:39.764216387 +0200
+++ samba-3.0.25a/source/script/installbin.sh.in	2007-05-26 07:46:32.740638424 +0200
@@ -21,13 +21,15 @@
  chmod $INSTALLPERMS $DESTDIR/$BINDIR/$p2
 
  # this is a special case, mount needs this in a specific location
- if [ $p2 = smbmount ]; then
-   if [ ! -d $DESTDIR/@rootsbindir@ ]; then
-      mkdir $DESTDIR/@rootsbindir@
-   fi
-   echo "Creating sym link $DESTDIR/@rootsbindir@/mount.smbfs to $BINDIR/$p2 "
-   ln -sf $BINDIR/$p2 $DESTDIR/@rootsbindir@/mount.smbfs
- fi
+ # Commented out for the Debian Samba package. We take care of this
+ # important symlink in debian/rules. (peloy@debian.org)
+ # if [ $p2 = smbmount ]; then
+ #  if [ ! -d $DESTDIR/@rootsbindir@ ]; then
+ #     mkdir $DESTDIR/@rootsbindir@
+ #  fi
+ #  echo "Creating sym link $DESTDIR/@rootsbindir@/mount.smbfs to $BINDIR/$p2 "
+ #  ln -sf $BINDIR/$p2 $DESTDIR/@rootsbindir@/mount.smbfs
+ # fi
 done
 
 
